src/Template/Element/wall/message.ctp

```diff
@@ -0,0 +1,162 @@
+<?php
+use App\Lib\LanguagesLib;
+use App\Model\CurrentUser;
+
+$username = isset($message->user) ? $message->user->username : CurrentUser::get('username');
+$avatar = isset($message->user) ? $message->user->image : CurrentUser::get('image');
+$createdDate = $message->date;
+$modifiedDate = $message->modified;
+$messageId = $message->id;
+$authorId = $message->owner;
+$messageText = $message->content;
+$messageHidden = $message->hidden;
+$sentence = null;
+$sentenceOwnerLink = null;
+
+$editUrl = $this->Url->build(array(
+    'controller' => 'sentence_comments',
+    'action' => 'edit',
+    $messageId
+));
+
+if (empty($modifiedDate) || $createdDate == $modifiedDate) {
+    $dateLabel = $this->Date->ago($createdDate);
+    $fullDateLabel = $createdDate;
+} else {
+    $dateLabel = format(
+        __('{createdDate}, edited {modifiedDate}'),
+        array(
+            'createdDate' => $this->Date->ago($createdDate),
+            'modifiedDate' => $this->Date->ago($modifiedDate)
+        )
+    );
+    $fullDateLabel = format(
+        __('{createdDate}, edited {modifiedDate}'),
+        array(
+            'createdDate' => $createdDate,
+            'modifiedDate' => $modifiedDate
+        )
+    );
+}
+$canViewContent = CurrentUser::isAdmin() || CurrentUser::get('id') == $authorId;
+$userProfileUrl = $this->Url->build(array(
+    'controller' => 'user',
+    'action' => 'profile',
+    $username
+));
+
+if (isset($message['Permissions'])) {
+    $menu = $this->Wall->getMenuFromPermissions(
+        $message,
+        $message['Permissions']
+    );
+} else {
+    $menu = [];
+}
+$menu[] = [
+    'text' => 'permalink',
+    'icon' => 'link',
+    'url' => [
+        'controller' => 'wall',
+        'action' => 'show_message',
+        $messageId,
+        '#' => "message_" .$messageId
+    ]
+];
+
+$children = $message->children;
+$cssClass = isset($isRoot) ? 'wall-thread' : 'reply';
+$canReply = false;
+?>
+
+<md-card id="message_<?= $messageId ?>" class="comment <?= $cssClass ?> <?= $messageHidden ? 'inappropriate' : '' ?>">
+    <md-card-header>
+        <md-card-avatar>
+            <?= $this->Members->image($username, $avatar, array('class' => 'md-user-avatar')); ?>
+        </md-card-avatar>
+        <md-card-header-text>
+            <span class="md-title">
+                <a href="<?= $userProfileUrl ?>"><?= $username ?></a>
+            </span>
+            <span class="md-subhead ellipsis" title="<?= $fullDateLabel ?>">
```

Comment:
A few days ago I've come across #1221 and since you use Material tooltips already on line 106 for all the menu buttons I think you should also use them here.

Similarly for https://github.com/Tatoeba/tatoeba2/blob/923b1a726e9db2831c7da8b167307a3d1e60e16d/src/Template/Element/sentence_comments/edit_form.ctp#L50 and https://github.com/Tatoeba/tatoeba2/blob/923b1a726e9db2831c7da8b167307a3d1e60e16d/src/Template/Element/wall/edit_form.ctp#L45